### ISSUE TITLE
Incorrect received date format?

### ISSUE THREAD

**Issue by axllent** [mhale/smtpd#42]
Hi @mhale - I was just just looking through the code in relation to something and noted that [you specify](https://github.com/mhale/smtpd/blob/master/smtpd.go#L795C33-L795C34) the received header date format as `now := time.Now().Format("Mon, _2 Jan 2006 15:04:05 -0700 (MST)")`. I'm pretty sure the `_2 Jan` is supposed to be `02 Jan` else dates < 10 end up being prefixed with a space, resulting in two spaces eg: `Mon,  5 Jan...` rather than `Mon, 05 Jan...`.

I could not find any exact reference in the RFC, but looking at the likes of Gmail for example they definitely prefix it with a `0` and not an additional space.

Interested in your feedback.

**Comment by mhale** [comment]
Hi, I've had a look and [RFC 2822 Section 3.3](https://datatracker.ietf.org/doc/html/rfc2822#section-3.3) says the day of the month can be 1 or 2 digits, and "the numeric day-of-month MUST be between 1 and the number of days allowed for the specified month (in the specified year)". I interpret that as no leading zeroes. I agree that the extra space is less than ideal and probably breaks a parser somewhere. If you feel strongly enough about it to send a PR, I'll merge it.

**Comment by axllent** [comment]
I believe you're absolutely correct regarding the leading `0` - ie: not being a requirement. I actually noticed that Google use both variations within the same headers on a single message while routing within their own infrastructure (go figure). I do think the additional space (when date < 10) should be removed though, so I've submitted a PR. No rush, just good to get in there at some point. Thanks mate.
